cli/compose/convert/volume.go

```diff
@@ -88,3 +67,77 @@ func convertVolumeToMount(
 	// Named volumes
 	return result, nil
 }
+
+func handleBindToMount(
+	volume composetypes.ServiceVolumeConfig,
+	result mount.Mount,
+) (mount.Mount, error) {
+	if volume.Volume != nil {
+		return result, errors.New("volume options are incompatible with type bind")
+	}
+	if volume.Tmpfs != nil {
+		return result, errors.New("tmpfs options are incompatible with type bind")
+	}
+	if volume.Bind != nil {
+		result.BindOptions = &mount.BindOptions{
+			Propagation: mount.Propagation(volume.Bind.Propagation),
+		}
+	}
+	// Binds volumes
+	return result, nil
+}
+
+func handleTmpfsToMount(
+	volume composetypes.ServiceVolumeConfig,
+	result mount.Mount,
+) (mount.Mount, error) {
+	// source must be empty for tmpfs
+	if volume.Source != "" {
+		return result, errors.New("invalid tmpfs source, source must be empty")
+	}
+	if volume.Bind != nil {
+		return result, errors.New("bind options are incompatible with type tmpfs")
+	}
+	if volume.Volume != nil {
+		return result, errors.New("volume options are incompatible with type tmpfs")
+	}
+	if volume.Tmpfs != nil {
+		result.TmpfsOptions = &mount.TmpfsOptions{
+			SizeBytes: volume.Tmpfs.Size,
+		}
+	}
+	// Tmpfs volumes
+	return result, nil
+}
+
+func convertVolumeToMount(
+	volume composetypes.ServiceVolumeConfig,
+	stackVolumes volumes,
+	namespace Namespace,
+) (mount.Mount, error) {
+	result := mount.Mount{
```

Comment:
I'm not a big fan of moving this outside of the `handleXX` functions; I assume you kept it out to reduce code duplication, but  now the conversion (and validation) is "split" between the `handleXX` functions and this function.

Given that not all mount-types use the same options in this type (e.g., `type=bind` uses `Source`, but `type=tmpfs` doesn't), I'd move this code to each `handleXX` function. If duplication of code is a concern (but I think it's minimal), you _could_ have a helper function for the shared code